CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/CRFHeartRateSampleProcessor.swift

```diff
@@ -59,16 +249,223 @@ internal class CRFHeartRateSampleProcessor {
     // MARK: Code ported from R
     // - note: R is indexed from 1 so to simplify porting the code, everything is offset by one
     // in the filters before calculating.
+    
+    func calculateHeartRate(_ input: [Double], samplingRate: Double) -> (heartRate: Double, confidence: Double) {
+        let window = Int(ceil(CRFHeartRateWindowSeconds * samplingRate))
+        guard let filtered = getFilteredSignal(input, samplingRate: Int(round(samplingRate))),
```

Comment:
Seems like you could save a heck of a lot of processing power by bandpass-filtering the signal as you go and passing in the filtered samples for the window of interest here, instead of rerunning the filter on the raw input samples for the whole window each time. Maybe I'm just not grokking the algorithm y'alls are using though. Is it not using a sliding 10 second window?

Reply:
Yup, next step. First I need to implement on Android. :)
I didn't want to introduce trying to test for that this iteration b/c only just finished up getting parity with the R implementation.

Reply:
Fair enough.